sql/src/main/java/io/crate/metadata/settings/session/SessionSettingRegistry.java

```diff
@@ -104,7 +109,34 @@
                     DataTypes.STRING.getName()
                 )
             )
-            .build();
+        .putAll(generateOptimizerRuleEntries())
+        .build();
+
+    private static Map<String, SessionSetting<?>> generateOptimizerRuleEntries() {
```

Comment:
I'm wondering if it would be worth to design a proper extension point. So that the `SessionSettingRegistry` isn't aware of all the concrete settings, but rather we've an extension point that other components can implement, and this registry here will discover them (e.g. via ServiceLoader). 


Might be over-engineering, but considering the module restructuring work (https://github.com/crate/crate/issues/9888) I think one question to ask is: Should a plugin or module *outside* of the `server` module be configurable via session settings? I think I'd answer that with yes, so an extesion point would be justified. And this concrete rule coupling also feels a bit wrong. 


Maybe @seut has some inputs?

Reply:
I agree that the rule coupling is not good. I like the idea of the discovery. Maybe there is a way  to do this with guice to keep things simple.

Reply:
> Maybe there is a way to do this with guice to keep things simple.

I'd avoid using guice. I think we should rather move away from it. We already use the ServiceLoader to load enterprise extensions. Lucene uses the service loader as well to load posting formats and others.

Reply:
ok, I was not aware that we are already using the ServiceLoader.